src/crosschain.ts

```diff
@@ -0,0 +1,112 @@
+import debug from 'debug'
+import { ethers } from 'ethers'
+import Web3 from 'web3'
+
+import { createDefaultClient } from './helpers'
+import { Address, LocalAddress, Client, Contracts, EthersSigner } from '.'
+
+import { AddressMapper } from './contracts/address-mapper'
+
+const log = debug('crosschain')
+
+export class CrossChain {
+  private _wallet: ethers.Signer
+  private _client: Client
+  private _address: Address
+  private _ethAddress: string
+  private _dappchainMapper: Contracts.AddressMapper
+
+  static async createOfflineCrossChainUserAsync(
+    endpoint: string,
+    privateKey: string,
+    dappchainEndpoint: string,
+    dappchainKey: string,
+    chainId: string
+  ): Promise<CrossChain> {
+    const provider = new ethers.providers.JsonRpcProvider(endpoint)
+    const wallet = new ethers.Wallet(privateKey, provider)
+    return CrossChain.createCrossChainUserAsync(wallet, dappchainEndpoint, dappchainKey, chainId)
+  }
+
+  static async createMetamaskCrossChainUserAsync(
```

Comment:
I think you need to merge in some of the eth-signer stuff from master.

Reply:
yeah will do on the base branch

Reply:
yeah will do on the base branch